ARCHITECTURE.md

```diff
@@ -0,0 +1,275 @@
+# HyperDB Architecture
+
+HyperDB is a scalable peer-to-peer key-value database.
+
+## Filesystem metaphor
+
+HyperDB is structured to be used much like a traditional hierarchical
+filesystem. A value can be written and read at locations like `/foo/bar/baz`,
+and the API supports querying or tracking values at subpaths, like how watching
+for changes on `/foo/bar` will report both changes to `/foo/bar/baz` and also
+`/foo/bar/19`.
+
+## Set of append-only logs
+
+A HyperDB is fundamentally a set of
+[hypercore](https://github.com/mafintosh/hypercore)s. A *hypercore* is a secure
+append-only log that is identified by a public key, and can only be written to
+by the holder of the corresponding private key. Because it is append-only, old
+values cannot be deleted nor modified. Because it is secure, any other peer in
```

Comment:
why is it secure?

Reply:
A feed can be downloaded from even untrustworthy peers and verified to be accurate. Any modifications (malicious or otherwise) to the original feed data by someone other than the author can be readily detected.